dll/win32/shell32/CDefView.cpp

```diff
@@ -1193,6 +1197,30 @@ void CDefView::ColumnListChanged()
     m_ListView.InvalidateRect(NULL, TRUE);
 }
 
+HRESULT CDefView::CompareIDsWithFallback(LPARAM Rule, PCUIDLIST_RELATIVE pidl1, PCUIDLIST_RELATIVE pidl2)
+{
+    UINT Flags = m_pSF2Parent ? (Rule & SHCIDS_BITMASK) : 0;
+    HRESULT hr = m_pSFParent->CompareIDs(Rule, pidl1, pidl2);
+    if (hr != 0)
+        return hr;
+    if (hr == 0 && m_sortInfo.DefaultFolderColumn > 0)
+        hr = m_pSFParent->CompareIDs(m_sortInfo.DefaultFolderColumn | Flags, pidl1, pidl2);
+    if (hr == 0 && (Rule & SHCIDS_COLUMNMASK) != 0)
+        hr = m_pSFParent->CompareIDs(0 | Flags, pidl1, pidl2);
```

Comment:
Why using `0 | Flags` (here and below), instead just `Flags` ?

Reply:
0 is the column. This LPARAM is column + flags and I'm just being specific about the column.

Reply:
Ok, makes sense this way.